### DISCUSSION TITLE
CSV-Data in Visual Studio Code not recognized when using other spacing characters

### DISCUSSION THREAD

**Discussion by hammerandpick** [microsoft/SandDance#606] [Ideas]
US Number Format is 1,000.00. In Germany the common format for numbers is 1.000,00. 
Therefore CSV Format uses ";" as seperator.
`Datum;Luftfeuchtigkeit;Regenzustand;Helligkeit;Sonnenschein;Sonnenscheindauer`
This is not recognized in SandDance and cannot be configured in VisualStudio Code.

The suggest is to implement configurable delimiters for spacing and as decimal point.   

See also issue #261. That has been solved for tabulator seperatros when naming files as tsv.

**Comment by danmarshall** [comment]
Perhaps we can look at the first line of the file (the headers) and see if it contains semicolons or commas?
It would be nice to have this automatic and not need UI or settings.

**Reply by hammerandpick** [reply]
I don't know if that is feasable, since there is a chance that someone used a semi-colon for commenting, thus creating a failure when importing such a file. Counting delimiters on each line might also not work properly because someone might leave out delimiters of emtpy rows at the end of the line.

**Reply by ShinySky42** [reply]
or we could do the lazy Excel fix with an annoncer on the first line until a proper solution is implemented ?
`sep=`
